net/acme/files/run.sh

```diff
@@ -129,9 +130,9 @@ pre_checks()
 
 post_checks()
 {
-    if [ -f "$STATE_DIR/post-acme.user" ]; then
-        log "Running user-provided cleanup script from $STATE_DIR/post-acme.user."
-        . "$STATE_DIR/post-acme.user" || return 1
+    if [ -n "$USER_CLEANUP" ] && [ -f "$USER_CLEANUP" ]; then
+        log "Running user-provided cleanup script from $USER_CLEANUP."
+        . "$USER_CLEANUP" || return 1
```

Comment:
Sourcing the script means it can modify the variables in the main run.sh script. That is bound to lead to errors; better to execute the script in a subprocess (i.e., replace '. "$USER_CLEANUP"' with just "$USER_CLEANUP").

Reply:
I added commit changing the logic to run the scripts in a subshell instead of sourcing like suggested.